hs-bindgen/src/HsBindgen/Backend/PP.hs

```diff
@@ -111,6 +118,31 @@ resolveGlobal = \case
     Storable_peek        -> importQ iForeign "peek"
     Storable_poke        -> importQ iForeign "poke"
     Foreign_Ptr          -> importQ iForeign "Ptr"
+
+    Eq_eq           -> importQ iPrelude "=="
+    Eq_uneq         -> importQ iPrelude "/="
+    Ord_lt          -> importQ iPrelude "<"
+    Ord_le          -> importQ iPrelude "<="
+    Ord_gt          -> importQ iPrelude ">"
+    Ord_ge          -> importQ iPrelude ">="
+    Base_identity   -> importQ iPrelude "id"
+    Base_not        -> importQ iPrelude "not"
+    Base_and        -> importQ iPrelude "&&"
+    Base_or         -> importQ iPrelude "||"
+    Bits_shiftL     -> importQ iBits    "shiftL"
+    Bits_shiftR     -> importQ iBits    "shiftR"
+    Bits_and        -> importQ iBits    ".&."
+    Bits_xor        -> importQ iBits    "xor"
+    Bits_or         -> importQ iBits    ".|."
+    Bits_complement -> importQ iBits    "complement"
+    Num_negate      -> importQ iPrelude "negate"
+    Num_add         -> importQ iPrelude "+"
+    Num_minus       -> importQ iPrelude "-"
+    Num_times       -> importQ iPrelude "*"
+    Fractional_div  -> importQ iPrelude "/"
+    Integral_rem    -> importQ iPrelude "rem"
```

Comment:
I think we are missing integral `quot`. Maybe there's a note somehwere, but `#define DIV(x,y) x / y` is a valid in integral context in C too. And often is.

Reply:
Very good point, I hadn't considered that at all. I will have to think about what the type of `/` should be.

Reply:
I will tackle this separately; see #277.